src/websocket/websocket.c

```diff
@@ -360,18 +358,18 @@ web_socket_util_parse_headers (const gchar *data,
               break;
             }
 
-          name = g_strndup (data, colon - data);
+          g_autofree gchar *name = g_strndup (data, colon - data);
           g_strstrip (name);
-          value = g_strndup (colon + 1, line - (colon + 1));
+          g_autofree gchar *value = g_strndup (colon + 1, line - (colon + 1));
           g_strstrip (value);
-          g_hash_table_insert (parsed_headers, name, value);
 
           if (!is_valid_line (name, -1) || !g_utf8_validate (value, -1, NULL))
             {
               g_debug ("received invalid header");
               consumed = -1;
               break;
             }
+          g_hash_table_insert (parsed_headers, g_steal_pointer(&name), g_steal_pointer(&value));
```

Comment:
This looks exactly correct, but a minor whitespace nag here.

```suggestion
          g_hash_table_insert (parsed_headers, g_steal_pointer (&name), g_steal_pointer (&value));
```